src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -609,22 +609,26 @@ static void CloseRequestResponseSession(RequestResponseAmqpLink requestResponseA
                         {
                             entry[ManagementConstants.Properties.ViaPartitionKey] = message.ViaPartitionKey;
                         }
-                }
+                    }
 
                     request.Map[ManagementConstants.Properties.Messages] = new List<AmqpMap> { entry };
 
-                    IEnumerable<long> sequenceNumbers = null;
                     var response = await this.ExecuteRequestResponseAsync(request).ConfigureAwait(false);
                     if (response.StatusCode == AmqpResponseStatusCode.OK)
                     {
-                        sequenceNumbers = response.GetValue<long[]>(ManagementConstants.Properties.SequenceNumbers);
+                        var sequenceNumbers = response.GetValue<long[]>(ManagementConstants.Properties.SequenceNumbers);
+                        if (sequenceNumbers == null || sequenceNumbers.Length < 1)
+                        {
+                            throw new ServiceBusException(true, "Could not schedule message successfully.");
+                        }
+
+                        return sequenceNumbers[0];
```

Comment:
Is there a chance that there's be more than a single sequence number?

Reply:
No. Should not happen.